Deploy pull-requests to a preview environment in fly.io

philter87 opened the pull request:
## Description

- This PR add deployment to seedcase-sprout-**preview** 
- This is relevant for reviewing a pull-request

I know that this implementation is a bit flawed since there is only one pull-request environment and several PRs will compete to override this environment. I have added a manual trigger as well to partly overcome this problem, but we might need to create a better approach in the future.

The url is:
https://seedcase-sprout-preview.fly.dev/

## Related Issues

<!-- List issues the PR closes -->

Closes #114.

## Testing

- [ ] Yes
- [X] No, not needed (give a reason below)
- [ ] No, I need help writing them

I am only able to test when created

## Reviewer Focus

APP_NAME defines if we deploy to seedcase-sprout or seedcase-sprout-pr. The app is deployed to seedcase-sprout when current_branch=main and otherwise seedcase-sprout-pr

## Checklist

<!-- This is to help you determine if your work is ready to be reviewed, if an item is not relevant then you can mark it as done (because you have checked and found that it isn't needed) -->

For all PRs that are not general documentation

- [ ] Tests accompany or reflect changes to the code
- [ ] Tests ran and passed locally
- [ ] Ran the linter and formatter
- [ ] Build has passed locally
- [X] Relevant documentation has been updated

pchmia committed:
ci: deploy pull-requests to seedcase-sprout-pr

pchmia committed:
fix: Using FLY_APP variable

pchmia committed:
fix: hardcode seedcase-sprout-pr test

pchmia committed:
fix: only seedcase-sprout

pchmia committed:
fix: rename to seedcase-sprout-preview which is more clear

signekb commented:
The PR with the newest will be on the url ... can we get a url for each PR? But then it should be closed when the PR is merged?

lwjohnst86 committed:
ci: :construction_worker: Use fly preview-pr-apps action

This action seems to deploy individual actions itself, as well as destroying them afterwards :D

lwjohnst86 committed:
ci: :construction_worker: Set the default name to the fly preview

lwjohnst86 committed:
ci: :rewind: Using the flyio preview action, so reverting to previous action

lwjohnst86 committed:
docs: :truck: This text makes more sense in the team docs rather than in the readme

philter87 commented:
Amazing @lwjohnst86. Way better!
